lib/cutting_edge/workers/dependency.rb

```diff
@@ -28,14 +29,66 @@ def perform(identifier, lang, locations, dependency_types, to_addr, auth_token =
       add_to_store(identifier, dependencies) unless @nothing_changed
     ensure
       unless @nothing_changed
+      #  alt_diff(dependencies[:locations], old_dependencies[:locations])
         badge_worker(identifier)
-        mail_worker(identifier, to_addr) if to_addr
+        mail_worker(identifier, to_addr, alt_diff(old_dependencies[:locations], dependencies[:locations])) if to_addr && !old_dependencies.empty?
       end
       GC.start
     end
   end
 
   private
+  
+  def alt_diff(old, new)
+    old_merged = old.values.first.merge(*old.values[1..-1]) {|key, current_val, new_val| current_val + new_val}
```

Comment:
I don't quite understand yet why we'd want to merge all dependencies from each location. Wouldn't we rather want a diff per location?